src/bluesky/tests/test_run_engine.py

```diff
@@ -1956,6 +1957,41 @@ def plan():
             RE(plan())
 
 
+def test_wait_with_returns_status(RE):
+    async def passing_future():
+        await asyncio.sleep(0.01)
+        return 1080
+
+    async def failing_future():
+        raise RuntimeError
+
+    async def timeout_future():
+        await asyncio.sleep(0.2)
+
+    timeout_task = asyncio.Task(timeout_future())
+
+    def runtime_error_and_success_plan():
+        statuses = yield from wait_for([lambda: passing_future(), lambda: failing_future()], timeout=0.1)
+        assert isinstance(statuses, set) and len(statuses) == 2
+        passing_future_status = [x for x in statuses if x.get_coro().__name__ == "passing_future"][0]
+        failing_future_status = [x for x in statuses if x.get_coro().__name__ == "failing_future"][0]
+        assert passing_future_status.done() and not passing_future_status.exception()
+        assert passing_future_status.result() == 1080
+        assert failing_future_status.done() and failing_future_status.exception()
```

Comment:
Can we check that task references passed through are maintained please?

```suggestion
        passing_task = asyncio.create_task(passing_future())
        failing_task = asyncio.create_task(failing_future())
        tasks = yield from wait_for([lambda: passing_task, lambda: failing_task], timeout=0.1)
        assert tasks == {passing_task, failing_task}
        assert passing_task.done() and not passing_task.exception()
        assert passing_task.result() == 1080
        assert failing_task.done() and failing_task.exception()
```